sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public virtual System.Net.ClientModel.Core.ErrorBehavior ErrorBehavior { get { throw null; } set { } }
+        public virtual void Apply(System.Net.ClientModel.Core.ClientMessage message) { }
+    }
+    public abstract partial class Result
+    {
+        protected Result() { }
+        public static System.Net.ClientModel.NullableResult<T> FromNullableValue<T>(T? value, System.Net.ClientModel.Core.MessageResponse response) { throw null; }
+        public static System.Net.ClientModel.Result FromResponse(System.Net.ClientModel.Core.MessageResponse response) { throw null; }
+        public static System.Net.ClientModel.Result<T> FromValue<T>(T value, System.Net.ClientModel.Core.MessageResponse response) { throw null; }
+        public abstract System.Net.ClientModel.Core.MessageResponse GetRawResponse();
+    }
+    public partial class Result<T> : System.Net.ClientModel.NullableResult<T>
+    {
+        internal Result() { }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool HasValue { get { throw null; } }
+        public override T Value { get { throw null; } }
+    }
+    public partial class UnsuccessfulRequestException : System.Exception
+    {
+        public UnsuccessfulRequestException(System.Net.ClientModel.Core.MessageResponse response) { }
+        protected UnsuccessfulRequestException(System.Net.ClientModel.Core.MessageResponse response, string message, System.Exception? innerException) { }
+        protected UnsuccessfulRequestException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
+        public int Status { get { throw null; } }
+    }
+}
+namespace System.Net.ClientModel.Core
+{
+    public partial class ClientMessage : System.IDisposable
+    {
+        protected internal ClientMessage(System.Net.ClientModel.Core.MessageRequest request) { }
+        public virtual System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
+        public bool HasResponse { get { throw null; } }
+        public virtual System.Net.ClientModel.Core.MessageClassifier MessageClassifier { get { throw null; } set { } }
+        public virtual System.Net.ClientModel.Core.MessageRequest Request { get { throw null; } }
+        public virtual System.Net.ClientModel.Core.MessageResponse Response { get { throw null; } protected internal set { } }
+        public virtual void Dispose() { }
+        protected virtual void Dispose(bool disposing) { }
+        public void SetProperty(System.Type type, object value) { }
+        public bool TryGetProperty(System.Type type, out object? value) { throw null; }
+    }
+    [System.FlagsAttribute]
+    public enum ErrorBehavior
+    {
+        Default = 0,
+        NoThrow = 1,
+    }
+    public partial interface IJsonModel<out T> : System.Net.ClientModel.Core.IModel<T>
```

Comment:
Did we consider adding `bool SupportsWrite` to avoid having to try/catch calls to `Write` where it is not supported like [here](https://github.com/Azure/azure-sdk-for-net/pull/39135/files#diff-6067a6456ce204bdf018695b964d58d82eaa0a9f901cd75b41e529fd8305b0fcR58)

Reply:
@m-nash FYI